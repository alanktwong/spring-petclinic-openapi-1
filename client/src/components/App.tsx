import * as React from 'react';

import { RouteComponentProps, withRouter } from 'react-router-dom';

import Menu from './Menu';

interface IAppProps extends RouteComponentProps {
  children: React.ReactNode;
}

interface IAppState {}

class App extends React.Component<IAppProps, IAppState> {
  render() {
    return (
      <div>
        <Menu name={this.props.location.pathname} />
        <div className="container-fluid">
          <div className="container xd-container">
            {this.props.children}

            <div className="container">
              <div className="row">
                <div className="col-12 text-center">
                  <img
                    src="/images/spring-pivotal-logo.png"
                    alt="Sponsored by Pivotal"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(App);
