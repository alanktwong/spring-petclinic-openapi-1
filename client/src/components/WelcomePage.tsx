import * as React from 'react';

export default () => (
  <div>
    <h2>Welcome</h2>
    <div className="row">
      <div className="col-md-12">
        <img className="img-responsive" src="/images/pets.png" alt="" />
      </div>
    </div>
  </div>
);
