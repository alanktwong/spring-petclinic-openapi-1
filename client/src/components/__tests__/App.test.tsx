import React from 'react';
import renderer from 'react-test-renderer';
import WelcomePage from '../WelcomePage';
import App from '../App';
import { StaticRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <App>
          <WelcomePage />
        </App>
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
