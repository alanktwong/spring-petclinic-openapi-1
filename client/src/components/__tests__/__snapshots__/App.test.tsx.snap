// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-default"
    role="navigation"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <span />
        </a>
        <button
          className="navbar-toggle"
          data-target="#main-navbar"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
      </div>
      <div
        className="navbar-collapse collapse"
        id="main-navbar"
      >
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li
            className="active"
          >
            <a
              href="/"
              title="home page"
            >
              <span
                aria-hidden="true"
                className="glyphicon glyphicon-home"
              />
               
              <span>
                Home
              </span>
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="/owners/list"
              title="find owners"
            >
              <span
                aria-hidden="true"
                className="glyphicon glyphicon-search"
              />
               
              <span>
                Find owners
              </span>
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="/vets"
              title="veterinarians"
            >
              <span
                aria-hidden="true"
                className="glyphicon glyphicon-th-list"
              />
               
              <span>
                Veterinarians
              </span>
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="/error"
              title="trigger a RuntimeException to see how it is handled"
            >
              <span
                aria-hidden="true"
                className="glyphicon glyphicon-warning-sign"
              />
               
              <span>
                Error
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container-fluid"
  >
    <div
      className="container xd-container"
    >
      <div>
        <h2>
          Welcome
        </h2>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <img
              alt=""
              className="img-responsive"
              src="/images/pets.png"
            />
          </div>
        </div>
      </div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-12 text-center"
          >
            <img
              alt="Sponsored by Pivotal"
              src="/images/spring-pivotal-logo.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
